apps/perudo_web/web/controllers/user_controller.ex

```diff
@@ -0,0 +1,67 @@
+defmodule PerudoWeb.UserController do
+  use PerudoWeb.Web, :controller
+
+  plug :authenticate when action in [:index, :show]
+
+  alias PerudoWeb.User
+
+  def index(conn, _params) do
+    users = Repo.all(User)
+    render(conn, "index.html", users: users)
+  end
+
+  def new(conn, _params) do
+    changeset = User.changeset(%User{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, %{"user" => user_params}) do
+    changeset = User.registration_changeset(%User{}, user_params)
+
+    case Repo.insert(changeset) do
+      {:ok, user} ->
+        conn
+        |> PerudoWeb.Auth.login(user)
+        |> put_flash(:info, "#{user.name} created successfully.")
+        |> redirect(to: user_path(conn, :index))
+      {:error, changeset} ->
+        render(conn, "new.html", changeset: changeset)
+    end
+  end
+
+  def show(conn, %{"id" => id}) do
+    user = Repo.get!(User, id)
+    render(conn, "show.html", user: user)
+  end
+
+  def edit(conn, %{"id" => id}) do
+    user = Repo.get!(User, id)
+    changeset = User.changeset(user)
+    render(conn, "edit.html", user: user, changeset: changeset)
+  end
+
+  def update(conn, %{"id" => id, "user" => user_params}) do
+    user = Repo.get!(User, id)
+    changeset = User.changeset(user, user_params)
+
+    case Repo.update(changeset) do
+      {:ok, user} ->
+        conn
+        |> put_flash(:info, "User updated successfully.")
+        |> redirect(to: user_path(conn, :show, user))
+      {:error, changeset} ->
+        render(conn, "edit.html", user: user, changeset: changeset)
+    end
+  end
+
+  defp authenticate(conn, _opts) do
```

Comment:
can we use a two-clause function for this? (ie matching on `(%{ assigns: { current_user } } = conn) when current_user` for the first, and `conn` for the second?